packages/blade/src/components/BottomSheet/__tests__/BottomSheet.web.test.tsx

```diff
@@ -0,0 +1,310 @@
+import { Badge } from '~components/Badge';
+import { Counter } from '~components/Counter';
+
+export const sleep = (delay = 10): Promise<number> =>
+  new Promise((resolve) => setTimeout(resolve, delay));
+
+const SingleSelectContent = (): React.ReactElement => {
+  return (
+    <ActionList>
+      <ActionListItem title="Home" value="Home" />
+      <ActionListItem title="Settings" value="settings" />
+      <ActionListItem title="Info" value="info" />
+      <ActionListItem title="Price" value="price" />
+      <ActionListItem title="Contact" value="contact" />
+      <ActionListItem title="About" value="about" />
+    </ActionList>
+  );
+};
+
+const MultiSelectContent = (): React.ReactElement => {
+  const fruites = [
+    'Apple',
+    'Apricot',
+    'Avocado',
+    'Banana',
+    'Orange',
+    'Blackberry',
+    'Blueberry',
+    'Cherry',
+    'Coconut',
+    'Cucumber',
+    'Durian',
+  ];
+
+  return (
+    <ActionList>
+      {fruites.map((fruit) => {
+        return <ActionListItem key={fruit} title={fruit} value={fruit} />;
+      })}
+    </ActionList>
+  );
+};
+
+beforeAll(() => jest.spyOn(console, 'error').mockImplementation());
+afterAll(() => jest.restoreAllMocks());
+
+describe('<BottomSheet />', () => {
+  const viewport = mockViewport({ width: '320px', height: '568px' });
+
+  it('should render Header/Footer/Body properly', () => {
+    const Example = (): React.ReactElement => {
+      return (
+        <BottomSheet isOpen={true}>
+          <BottomSheetHeader
+            title="Address Details"
+            subtitle="Saving addresses will improve your checkout experience"
+            trailing={<Badge variant="positive">Action Needed</Badge>}
+            titleSuffix={<Counter intent="positive" value={2} />}
+          />
+          <BottomSheetBody>
+            <Text>BottomSheet body</Text>
+          </BottomSheetBody>
+          <BottomSheetFooter>
+            <Button isFullWidth variant="secondary">
+              Remove address
+            </Button>
+          </BottomSheetFooter>
+        </BottomSheet>
+      );
+    };
+    const { container } = renderWithTheme(<Example />);
+    expect(container).toMatchSnapshot();
+  });
+
+  it('should open/close BottomSheet', async () => {
+    const user = userEvents.setup();
+
+    const Example = (): React.ReactElement => {
+      const [isOpen, setIsOpen] = React.useState(false);
+
+      return (
+        <>
+          <Button onClick={() => setIsOpen(true)}>Open</Button>
+          <BottomSheet isOpen={isOpen} onDismiss={() => setIsOpen(false)}>
+            <BottomSheetHeader title="Select Account" />
+            <BottomSheetBody>
+              <Text>BottomSheet body</Text>
+            </BottomSheetBody>
+          </BottomSheet>
+        </>
+      );
+    };
+    const { getByText, queryByTestId } = renderWithTheme(<Example />);
+
+    expect(queryByTestId('bottomsheet-body')).not.toBeInTheDocument();
```

Comment:
Can we query on the bottomsheet content instead? That would be closer to user behavior.

Reply:
Right. updated the ones which have text, but kept the ActionList examples to use testID only because we keep rendering those element just set opacity: 0 to the bottomsheet, I'm unable to do `expect(getByRole('combobox', { name: 'Select Action' })).not.toBeVisible();` seems like jest doesn't take into consideration that the parent element is invisible while calculating opacity.